packages/component-base/src/slot-controller.d.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { ReactiveController } from 'lit';
+
+export class SlotController implements ReactiveController {
+  constructor(
+    host: HTMLElement,
+    slotName: string,
+    slotFactory: () => HTMLElement,
+    slotInitializer: (host: HTMLElement, node: HTMLElement) => void
+  );
+
+  hostConnected(): void;
+
+  /**
+   * The controller host element.
+   */
+  host: HTMLElement;
+
+  /**
+   * The slotted node managed by the controller.
+   */
+  node: HTMLElement;
+
+  slotName: string;
+
+  initialized: boolean;
```

Comment:
Are these intentionally public? `initialized` kind of looks like an internal state, and the other ones are passed via the constructor. I'd imagine there's no need to access these properties from outside.

Reply:
Good catch, fixed. I might need `initialized` in the `HelperController` - changed it to be protected.

Reply:
Are `host` and `node` still needed in the public interface?

Reply:
We use `host` in other controllers e.g. `FocusTrapController` so I would keep it.
Regarding `node`: yes, I will need it later when adding `FlattenedNodesObserver`.